app/frontend/src/components/ches/UploadContexts.vue

```diff
@@ -0,0 +1,385 @@
+ * The contexts are also displayed to the user in a table
+ */
+
+import XLSX from 'xlsx';
+import moment from 'moment';
+import { mapActions } from 'vuex';
+import { mapFields } from 'vuex-map-fields';
+
+import { Contexts, Formats } from '@/utils/constants';
+import * as Utils from '@/utils/utils';
+
+export default {
+  name: 'UploadContexts',
+
+  data() {
+    return {
+      dragover: false,
+    };
+  },
+
+  computed: {
+    ...mapFields('ches', [
+      'mergeForm.excelParsed',
+      'mergeForm.excel',
+      'mergeForm.contexts',
+      'mergeForm.contextsType',
+      'mergeForm.contextVariables',
+    ]),
+  },
+
+  methods: {
+    ...mapActions('alert', ['showAlert', 'clearAlert']),
+
+    addExcelFile(e) {
+      this.dragover = false;
+
+      const file = e.dataTransfer.file;
+      // validate Excel file
+      if (!this.validateFile(file)) {
+        // show error alert
+        this.alert = {
+          type: 'warning',
+          text: 'Context file size cannot excede 20MB',
+        };
+        this.showAlert(this.alert);
+      }
+      // else attachments are valid
+      else {
+        // parse file
+        this.parseFile(file, this.addContextsToStore);
+        // make file data avalable in parent component
+        this.$emit('fileUploaded', file);
+        // show table of contexts
+        this.excelParsed = true;
+
+        this.clearAlert();
+      }
+    },
+
+    parseFile(file, cb) {
+      // this will suppress a console warning about moment deprecating a default fallback on non ISO/RFC2822 date formats
+      // we will just force it to use the new Date constructor.
+      moment.createFromInputFallback = function (config) {
+        config._d = new Date(config._i);
+      };
+
+      const reader = new FileReader();
+      const rABS = !!reader.readAsBinaryString;
+
+      reader.onload = (e) => {
+        /* Parse data */
+        const bstr = e.target.result;
+
+        const wb = XLSX.read(bstr, {
+          type: rABS ? 'binary' : 'array',
+          cellDates: true,
+          dateNF: Formats.EXCEL_PARSE_FORMAT,
+        });
+        /* Get first worksheet */
+        const wsname = wb.SheetNames[0];
+        const ws = wb.Sheets[wsname];
+        /* Convert array of arrays */
+        const data = XLSX.utils.sheet_to_json(ws, { header: 1, raw: false });
+
+        let excel = { cols: [], headers: [], data: [] };
+        excel.cols = this.make_cols(ws['!ref']);
+
+        const headers = this.sanitizeHeaders(data[0]);
+        excel.headers = [headers];
+        // actual data is rows 2 onward
+        excel.data = data.slice(1);
+
+        let contexts = [];
+
+        excel.data.forEach((d) => {
+          let r = {
+            to: [],
+            cc: [],
+            bcc: [],
+            tag: undefined,
+            delayTS: undefined,
+            delayTSDisplay: undefined,
+            context: {},
+          };
+          let fields = excel.cols;
+          fields.forEach((f) => {
```

Comment:
Since fields is never reassigned, and it's only really needed to reference an iteration, it'd be better to just directly call the object and save some memory with an unnecessary memory assignment:
```js
excel.cols.forEach((f) => {
  // ...
}
```